blog/2022-05-04-announcing-stryker-6-effortless-sauce-mutation.md

```diff
@@ -0,0 +1,257 @@
+
+1. Mutant A: we start a worker process and test it.
+1. Mutant B: we need to recreate the worker to test the static mutant
+1. Mutant C: we need to recreate the worker _again_, because the previous mutant was static, which polluted the worker with a non-erasable mutant.
+
+Basic math tells us a worker is started three times in this example. Now imagine this example again, but with the mutants sorted: A, C, and B.
+
+1. Mutant A: we start a worker process and test it.
+1. Mutant C: we reuse the worker and test it.
+1. Mutant B: we need to recreate the worker to test the static mutant
+
+As you can see, sorting the mutants just saved us from having to restart the worker once. This performance gain might sound insignificant, but with 1000s of mutants, it can save 100s of restarts, well worth the (small) effort.
+
+One final note about static mutants: the progress-reporter will now accurately report their progress. It does this by assigning a _weight_ to each mutant to be tested, which it calculates using the performance measurements of the dry run. This means more smooth progress and a more realistic ETC estimate.
+
+## 🖐 Ignore static mutants
+
+Testing static mutants can be very expensive. As explained in the previous paragraph, they need a fresh worker process for each run (for Node-based test runners). On top of that, Stryker cannot determine test coverage for them. So the only thing it can do is run all tests 🐌.
+
+Also, they are generally less interesting for you to spend your time on since they include constant strings, numbers, etc.
+
+You might even want to ignore them entirely! This is where `--ignoreStatic` comes in. With `--ignoreStatic` static mutants are ... well ignored. Of course, you can still see them in your HTML report, but they won't count towards your mutation score.
+
+<figure><img src="/images/blogs/ignored-static-mutant.png" alt="Ignored static mutant" />
+<figcaption>An ignored static mutant</figcaption>
+</figure>
+
+Ignoring static mutants improves performance, but your mileage may vary. It depends on the number of static mutants and the size of your project. For running StrykerJS on Stryker's core itself, we [saw a whopping 50% performance improvement by ignoring 6% of our mutants (the static mutants) 🤯](https://github.com/stryker-mutator/stryker-js/issues/3282#issuecomment-980110861).
+
+Even though it is a performance improvement, the default value for`"ignoreStatic"` is `false`, meaning Stryker won't ignore them. Our reasoning here is: "better be safe than sorry". But how would you know about this new feature when running Stryker on your project? Well, Stryker will tell you about it with a new warning message.
+
+You will see the message when 2 conditions are met:
+
+1. Stryker estimates static mutants to take >= 40% of the total execution time; and
+2. The average estimated execution time of static mutants in your project is twice that of non-static mutants.
```

Comment:
The way I read this (due to everything being plural) is that at least 66% of the time is spent testing static mutants (swince the execution of statis mutations in your project is twice that of non-static mutants). I think this would be clearer in the singular form:
`The average estimated execution time of a static mutant in your project is twice that of a non-static mutant.`